src-plugins/reformat/medReformatWorkspace.cpp

```diff
@@ -0,0 +1,75 @@
+#include "medReformatWorkspace.h"
+
+#include <medLayerParameterGroup.h>
+#include <medViewContainer.h>
+#include <medTabbedViewContainers.h>
+#include <medSettingsManager.h>
+#include <medToolBox.h>
+#include <medToolBoxFactory.h>
+#include <medWorkspaceFactory.h>
+#include <medViewContainerManager.h>
+#include <medViewEventFilter.h>
+#include <medViewParameterGroup.h>
+#include <medDataManager.h>
+#include <medAbstractSelectableToolBox.h>
+#include <medSelectorToolBox.h>
+
+#include <stdexcept>
+
+class medReformatWorkspacePrivate
+{
+public:
+    // Give values to items without a constructor.
+    medReformatWorkspacePrivate() :
+       selectorToolBox(NULL)
+    {}
+
+    medSelectorToolBox* selectorToolBox;
+};
+
+medReformatWorkspace::medReformatWorkspace(QWidget *parent) : medAbstractWorkspace(parent), d(new medReformatWorkspacePrivate)
+{
+    d->selectorToolBox = new medSelectorToolBox(parent, "reformat");
+    connect(d->selectorToolBox,SIGNAL(success()),this,SLOT(onSuccess()));
+
+    // Always have a parent.
+    if (!parent)
+    {
+        throw (std::runtime_error ("Must have a parent widget"));
+    }
```

Comment:
1.) If you shouldn't call the constructor with no parent, then the constructor shouldn't have a NULL default parent argument.
2.) This would not be a runtime error but a programming error, so should use assert(parent), placed at the top of the function.